### ISSUE TITLE
Crash when overflowing text inside a view with a height

### ISSUE THREAD

**Issue by SpudNyk** [diegomura/react-pdf#1923]
**Describe the bug**

When rendering text that overflows its container (even if it's overflow hidden) I get the following error:
```
worker.ts:38 TypeError: Cannot read properties of undefined (reading 'y')
    at textkit.es.js:787:1
    at Array.map (<anonymous>)
    at layoutLines (textkit.es.js:778:1)
    at textkit.es.js:825:1
    at textkit.es.js:877:1
    at compose.js:32:1
    at textkit.es.js:1748:19
    at layoutText (index.es.js:2085:1)
    at index.es.js:2926:1
```

This did not occur in version 2.1.1, but did in 2.3.0

**To Reproduce**
The following code will reproduce the error in the repl

```jsx
const styles = {
    container: {
        height: 30,
    },
    content: {
        overflow: 'hidden',
        textOverflow: 'ellipsis',
        maxLines: 1,
    },
};

function TextOverflowError() {
    return (
        <Document>
            <Page>
                <View style={styles.container}>
                    <Text style={styles.content}>
                        En un lugar de la Mancha, de cuyo nombre no quiero
                        acordarme, no ha mucho tiempo que vivía un hidalgo de
                        los de lanza en astillero, adarga antigua, rocín flaco y
                        galgo corredor. Una olla de algo más vaca que carnero,
                        salpicón las más noches, duelos y quebrantos los
                        sábados, lentejas los viernes, algún palomino de
                        añadidura los domingos, consumían las tres partes de su
                        hacienda. El resto della concluían sayo de velarte,
                        calzas de velludo para las fiestas con sus pantuflos de
                        lo mismo, los días de entre semana se honraba con su
                        vellori de lo más fino. Tenía en su casa una ama que
                        pasaba de los cuarenta, y una sobrina que no llegaba a
                        los veinte, y un mozo de campo y plaza, que así
                        ensillaba el rocín como tomaba la podadera. Frisaba la
                        edad de nuestro hidalgo con los cincuenta años, era de
                        complexión recia, seco de carnes, enjuto de rostro; gran
                        madrugador y amigo de la caza. Quieren decir que tenía
                        el sobrenombre de Quijada o Quesada (que en esto hay
                        alguna diferencia en los autores que deste caso
                        escriben), aunque por conjeturas verosímiles se deja
                        entender que se llama Quijana; pero esto importa poco a
                        nuestro cuento; basta que en la narración dél no se
                        salga un punto de la verdad
                    </Text>
                </View>
            </Page>
        </Document>
    );
}

ReactPDF.render(<TextOverflowError />);
```

[react-pdf REPL](https://react-pdf.org/repl?code=3187b0760ce02e004d09e01b02984605e181bc05037cca185008602598c804e017367818c016c990399351d013000c00d037c017d06322519313ab91a31000dda8033442003b9d00e44cc80133d52b58b9f92400f2801e4955551bb724488c80070864231a17202d897300190a543a004613114161006e1c1c6500573060283270180015644b1b15357500512a2a102a000a004a7a462a642844aa3018729f46001e0011106044df2928003e36d31876800512366461d1b9b1803532647538282464742c041408003a09724a2a6159f9f3f6eccb358dad9dd403f049625391f3b9c2e6e4984444b6122a0c10c3f120c000b2241493048fc207210889780806060102f800467514722008e8965a94deef5309140543d003fab0d44c1a1305f225804c645a590be572e312f08519014005bd077d7414c41b1918622713186a3430310908017a82a43012340c8ce6a08161240a551feaae23b1123098295807ce6aa949c8f824aa5f87fa8b9164ba9e8ca7b1800154c0a09033941c09233be90043b402949a0bc7c38000e3b6bed0e883075c64600019f9af2b42f9c3d8a66a1617a2e4ca60f0180c73190a8080d032a4d4a20a1f46eaebb09431190002b354fceb301e75128101d68a005fcd57306d114717211bc4920011f75fa46a82cbeedf050c5a3c2380207d01766fb503a9a1a754492cbe147c5fbda1c065980299ec2a20cda85adc2030794afc279c02412270a0e4e002922440ea8c71a204aadee0738893ba3015e9b9f6ca32cd01f6441c08925e35a24f62de0fb9c6a3d29e2f8dabf6b280a8840c588406ca42a0b31b4b8054090adbfef8591f3128ce19464181145e668161a8a7ad9180028c02a91e841aa82afaaabf895972fc5ccd38a63c5ca03af40080ca6856c91b120262143465cb623f0a0ff0f124169a31964a050507963037cd442af3929ecada687ca4aac231aaa101f2ce69852078018f14e19a3d25a07b51302d6fe0f1f29a120052861719e8006254278dc461515c8c82e5605805cb40a52d2fa30662af165b3e29172c42822bbb6f257170b95e21a2ae0a0e61905997ecfa9acc680605c64d116f2580dda243fb02a51d5200c4301b05c58003410fe1e8541fc6da02158903bb3572a827192a9e8008af8b502aa18cfa026173c02bedf802510059752a218962c0a3d642f6148c0c8a3da82eaa0b94614aadfb22d09da307cc4d59940990ca33d29190cab66038902b594a8069f0a18d02c66a880df7c910300c57a2522543ab246147280910ddbd41b8465a8453baec700533d93968dcc031487979322dd917682a0ef6be96dae16af4cfe64005d7a821c8cde2c4ba60d5dfbd5bd00c9b4c0ad8e1b2caa596fa25292637347a000979f8d2cc5d329a8a2a5a1c92adf0965b62e5e57b4003d15c433f1e1d2c2b19c171879334cf1dc8e1f74bd3f4c40a7951c4c23c40012b20a4940e3274055ec00de5e525c39358b63d845094650c061e0c39ce0400)

**Expected behavior**
A crash to not occur

**Comment by davbrito** [comment]
I'm having the same issue since the update to the latest version

**Comment by pganster** [comment]
We experience the same bug when upgrading from @react-pdf/renderer@2.1.1 to @react-pdf/renderer@2.1.2.

Additionally, when a Text is inside a View with a fixed height, whilst the text itself needs more height, the rendering crashes too.

**Comment by carlobeltrame** [comment]
I already fixed this in #1917. We just have to wait for a release of react-pdf which includes this.

**Comment by davbrito** [comment]
> I already fixed this in #1917. We just have to wait for a release of react-pdf which includes this.

When will this be released. I think it's critical.

**Comment by ads-silva** [comment]
Do we have any idea when a new version will be released?

**Comment by modosc** [comment]
any chance for a release here? it's been a month and we can't manually pull in this fix.

**Comment by carlobeltrame** [comment]
@diegomura seems to be on vacation or something. But the bug was introduced in @react-pdf/renderer v2.3.0, so you can pin to v2.2.0 for now until a new version is released.
By the way, @jeetiss you do not by any chance have the rights to publish new versions, do you?

**Comment by bjornbreck** [comment]
Any updates about when this will get published? @diegomura

**Comment by christopherromero** [comment]
@diegomura Any updates on this fix?

**Comment by carlobeltrame** [comment]
This was fixed and published with [@react-pdf/textkit@4.1.0](https://github.com/diegomura/react-pdf/releases). Does the fix not work for you @christopherromero? If it does, this issue could be closed.

**Comment by pganster** [comment]
After updating to `@react-pdf/renderer` v3.0.0 I can confirm that no more crashes occur, so I think this issue can be closed.
